labelbox/data/generator.py

```diff
@@ -64,19 +77,22 @@ def fill_queue(self):
                     raise ValueError("Unexpected None")
                 self.queue.put(value)
         except Exception as e:
-            logger.warning("Unexpected exception while filling the queue. %r",
-                           e)
-        finally:
-            self.queue.put(None)
```

Comment:
you still need this finally where you write a None to the queue to indicate that the job is complete.